src/app.ts

```diff
@@ -1 +1,23 @@
-console.log('hello world')
\ No newline at end of file
+import getIssuesByProject from './api-functions/get-issues-by-project'
+import BQ from './big-query-functions/big-query-service-class';
+import issueSchema from './schemas/issue-schema'
+
+async function pullIssuesAndCreateDatabase(datasetId: string, projectId: string) {
+    const issues = await getIssuesByProject(projectId);
+    const isTable = await BQ.tableExists(datasetId, projectId);
+
+    if (isTable) {
+        await BQ.deleteTable(datasetId, projectId);
+    }
+    
+    await BQ.createTable(datasetId, projectId, issueSchema);
+
+    console.log('Waiting 10 seconds for table to be accessible')
+    //Timeout set to avoid inconsistencies with server-side availability of newly created tables
+    setTimeout(async () => {
+        await BQ.insertRows(datasetId, projectId, issues)
+    }, 10000)
```

Comment:
Is this timeout necessary? will it not work without wait?

Reply:
So the problem faced here is that a newly created table may not appear to exist in the database until some time has passed. I found some discussion on the subject here: 
https://github.com/googleapis/google-cloud-go/issues/975

Sometimes the rows would insert no problem into the newly created table, and sometimes it would return a 404 error. It definitely appears to be a server-side issue.

There were a few ways I thought of fixing it:

1. Instead of deleting the table and creating a new table, instead delete the contents of the table and upload the new data. This at first glance seems to be the best solution, however it runs into a Streaming Buffer: where newly created records in the BQ storage cannot be deleted or edited until a 90 minute period has ended. This solution would work if you only ran the program once every 2 hours or longer, but it's hard to test, as you have to wait 90 minutes between tests, and this solution still doesn't cover the case of when a new table is created anyways.

2. Since we know we just created a new table, if insertRows returns a 404 error we could just run the function again until the rows are inserted or a different error is returned. The downside is this could cause a lot of BQ API calls. I have not tried this solution yet, but it seem feasible.

3. The solution I provided here with the timeout, which is maybe a little messy, but it was recommended in the above thread.

If you would like me to implement either of the other solutions let me know I can do that.

Reply:
Good investigation. For the time being, lets keep it as it is.